daemon/daemon.go

```diff
@@ -77,25 +78,42 @@ type Command struct {
 	// is this path accessible on the snapd-snap socket?
 	SnapOK bool
 
+	// can polkit grant access?
+	ActionID string
+
 	d *Daemon
 }
 
+var polkitCheckAuthorization = polkit.CheckAuthorization
+
 func (c *Command) canAccess(r *http.Request, user *auth.UserState) bool {
 	if user != nil {
 		// Authenticated users do anything for now.
 		return true
 	}
 
 	isUser := false
-	uid, err := ucrednetGetUID(r.RemoteAddr)
+	pid, uid, err := ucrednetGet(r.RemoteAddr)
 	if err == nil {
 		if uid == 0 {
 			// Superuser does anything.
 			return true
 		}
 
+		if c.ActionID != "" {
+			subject := polkit.ProcessSubject{Pid: int(pid)}
+			if result, err := polkitCheckAuthorization(subject, c.ActionID, nil, polkit.CheckAuthorizationAllowUserInteraction); err == nil {
```

Comment:
TBH I think I'd rather move make the polkit package handle all of this. That is, have something like a `polkit.IsPidAuthorized(pid int) (bool, error)`. Is there a reason you didn't go for that approach?